src/test/run-make/thumb-none-cortex-m/Makefile

```diff
@@ -0,0 +1,35 @@
+-include ../../run-make-fulldeps/tools.mk
+
+# How to run this
+# $ ./x.py clean
+# $ ./x.py test --target thumbv6m-none-eabi,thumbv7m-none-eabi src/test/run-make
+
+# Supported targets:
+# - thumbv6m-none-eabi (Bare Cortex-M0, M0+, M1)
+# - thumbv7em-none-eabi (Bare Cortex-M4, M7)
+# - thumbv7em-none-eabihf (Bare Cortex-M4F, M7F, FPU, hardfloat)
+# - thumbv7m-none-eabi (Bare Cortex-M3)
+
+# See https://stackoverflow.com/questions/7656425/makefile-ifeq-logical-or
+ifneq (,$(filter $(TARGET),thumbv6m-none-eabi thumbv7em-none-eabi thumbv7em-none-eabihf thumbv7m-none-eabi))
+
+# We need to be outside of 'src' dir in order to run cargo
+WORK_DIR := $(RUST_TEST_TMPDIR)/run-make/$(TARGET)
+
+CRATE := cortex-m
+CRATE_VER := 0.5.0
+
+RUSTC := $(RUSTC_ORIGINAL)
+LD_LIBRARY_PATH := $(HOST_RPATH_DIR)
+
+all:
+	env
+	mkdir -p $(WORK_DIR)
+	-cd $(WORK_DIR) && rm -rf $(CRATE)
+	cd $(WORK_DIR) && $(CARGO) clone $(CRATE) --vers $(CRATE_VER)
```

Comment:
`cargo clone` is not an internal command of cargo (rust-lang/cargo#1545), you'll need to install [`cargo-clone`](https://crates.io/crates/cargo-clone) on the CI to get it. And I don't think it is a good idea have an un-vendored in the test case.

Perhaps consider making `cortex-m` a submodule?

---

The typical way to test an external crate is via `cargotest`:

```sh
./x.py test src/tools/cargotest
```

The problem is that `cargotest` is only designed for running on self-hosted platforms. If you have time, you could modify `cargotest` to run in cross-platform:

1. In `src/tools/cargotest/main.rs`
    1. Edit `TEST_REPOS` and specify the expected target triples of each repository
    2. Edit `fn main()` to accept the target 
    3. Edit `fn run_cargo_test()` to pass `--target` to cargo
2. Edit `struct Cargotest` in `src/bootstrap/test.rs` to make it runnable when `host ≠ target`
3. Edit `src/ci/docker/dist-various-1/Dockerfile` to test `src/tools/cargotest` for the `$RUN_MAKE_TARGETS`.